src/components/MapView.js

```diff
@@ -73,9 +82,26 @@ class MapView extends Component {
         }
     }
 
+    handleClose = () => {
+        const { history} = this.props;
+        this.setState({legend: false}, () => {
+            history.push('/');})
+    };
+
+    showLegend = () => {
+        return speciesColorMap.entrySeq().map( ([key, value]) =>
```

Comment:
I think it'd be a little cleaner to write a `getAllSpecies` function in the `ColorService` and import/export that, then look up each color. Another approach would be to write a `getColorSpeciesPairs` that returns a list of pairs: `'[['black bear', color], ...]`, then iterate over those.